### PR TITLE
Feature/11870 fss tooling search for duplicate batches before executing job

### PR DESCRIPTION
**Pull request by Rashi14748**
This PR contains implementation and unit tests for search duplicate batches before executing new batch job.

### CODE REVIEW

File: file-share-service-desktop-client/Modules/Admin/JobViewModels/NewBatchJobViewModel.cs [hunk 1]
```diff
@@ -129,84 +134,95 @@ protected internal override async Task OnExecuteCommand()
...
-                FileUploadProgress.Clear();
-                try
+                var buildBatchModel = BuildBatchModel();
+                BatchSearchResponse batchSearchResponse = await SearchBatch();
+                logger.LogInformation($"{ batchSearchResponse.Count} duplicate batches found for action: {Action} and displayName: {DisplayName}.");
+                if (batchSearchResponse.Total > 0 && messageBoxService.ShowMessageBox($"Confirmation for displayName: {DisplayName}", $"{batchSearchResponse.Count} duplicate batches found. Do you still want to continue to execute the job ?",
+                MessageBoxButton.YesNo, MessageBoxImage.Question) == MessageBoxResult.No)
+                {
+                    logger.LogInformation($"File Share Service create new batch job cancelled for action: {Action} and displayName: {DisplayName}, because {batchSearchResponse.Count} duplicate batches found.");
+                    ExecutionResult = $"File Share Service create new batch cancelled to execute the job. ";
```
Comment (crdandoukho) [inline file-share-service-desktop-client/Modules/Admin/JobViewModels/NewBatchJobViewModel.cs:149]: Please change "File Share Service create new batch cancelled to execute the job. " to "File Share Service create new batch cancelled"

File: file-share-service-desktop-client/Modules/Admin/JobViewModels/NewBatchJobViewModel.cs [hunk 2]
```diff
@@ -396,22 +412,47 @@ private bool IsDirectoryExist(string directory)
...
+            return await fileShareClient.Search(filter, null, null);
+        }
+
+        private string buildSearchQuery()
+        {
+            var queryBuilder = new StringBuilder();
+
+            queryBuilder.Append($"businessunit eq '{BusinessUnit}'");
+
+            if (Attributes != null)
```
Comment (crdandoukho) [inline file-share-service-desktop-client/Modules/Admin/JobViewModels/NewBatchJobViewModel.cs:438]: Please also check `Attributes` is not empty

File: file-share-service-desktop-client/Modules/Admin/JobViewModels/NewBatchJobViewModel.cs [hunk 3]
```diff
@@ -129,84 +134,95 @@ protected internal override async Task OnExecuteCommand()
...
+                logger.LogInformation("Execute job started for Action : {Action} and displayName :{displayName} .", Action, DisplayName);
                 var fileShareClient = fileShareClientFactory();
-                var buildBatchModel = BuildBatchModel();                
-                logger.LogInformation("File Share Service batch create started.");
-                batchHandle = await fileShareClient.CreateBatchAsync(buildBatchModel, CancellationToken.None); 
-                logger.LogInformation("File Share Service batch create completed for batch ID:{BatchId}.", batchHandle.BatchId);
-                FileUploadProgress.Clear();
-                try
+                var buildBatchModel = BuildBatchModel();
+                BatchSearchResponse batchSearchResponse = await SearchBatch();
```
Comment (crdandoukho) [inline file-share-service-desktop-client/Modules/Admin/JobViewModels/NewBatchJobViewModel.cs:143]: Could you please ensure any exceptions raised by `SearchBatch()` are handled gracefully. At the moment an exception from `SearchBatch()` is caught by the global exception handler.

**Review by crdandoukho — CHANGES_REQUESTED** [review]
Some changes required please. 

I like the new MessageBoxService!

File: file-share-service-desktop-client/Modules/Admin/JobViewModels/NewBatchJobViewModel.cs (hunk 1 above)
Comment (Rashi14748) [reply file-share-service-desktop-client/Modules/Admin/JobViewModels/NewBatchJobViewModel.cs:149]: updated as suggested. Thanks

File: file-share-service-desktop-client/Modules/Admin/JobViewModels/NewBatchJobViewModel.cs (hunk 2 above)
Comment (Rashi14748) [reply file-share-service-desktop-client/Modules/Admin/JobViewModels/NewBatchJobViewModel.cs:438]: added empty check as well. Thanks

File: file-share-service-desktop-client/Modules/Admin/JobViewModels/NewBatchJobViewModel.cs (hunk 3 above)
Comment (Rashi14748) [reply file-share-service-desktop-client/Modules/Admin/JobViewModels/NewBatchJobViewModel.cs:143]: added try-catch block. Thanks

**Review by crdandoukho — APPROVED** [review]
Looks good!

**Review by barrie-cooper — APPROVED** [review]
LGTM

**Pull request merged**